src/main/java/dev/morling/onebrc/CalculateAverage_royvanrijn.java

```diff
@@ -15,65 +15,309 @@
+                    // Simple is faster for '\n' (just three options)
+                    int endPointer;
+                    if (bb.get(separatorPointer + 4) == '\n') {
+                        endPointer = separatorPointer + 4;
+                    }
+                    else if (bb.get(separatorPointer + 5) == '\n') {
+                        endPointer = separatorPointer + 5;
+                    }
+                    else {
+                        endPointer = separatorPointer + 6;
+                    }
 
+                    // Read the entry in a single get():
+                    bb.get(buffer, 0, endPointer - startPointer);
+                    bb.position(endPointer + 1); // skip to next line.
+
+                    // Extract the measurement value (10x):
+                    final int nameLength = separatorPointer - startPointer;
+                    final int valueLength = endPointer - separatorPointer - 1;
+                    final int measured = branchlessParseInt(buffer, nameLength + 1, valueLength);
+                    measurements.getOrCreate(buffer, nameLength).updateWith(measured);
+                }
+                return measurements;
+            }
+            catch (IOException e) {
+                throw new RuntimeException(e);
+            }
+        }).parallel().flatMap(v -> v.values.stream())
+                .collect(Collectors.toMap(e -> new String(e.key), BitTwiddledMap.Entry::measurement, (m1, m2) -> m1.updateWith(m2), TreeMap::new));
+
+        // Seems to perform better than actually using a TreeMap:
+        System.out.println(results);
     }
+
+    /**
+     * -------- This section contains SWAR code (SIMD Within A Register) which processes a bytebuffer as longs to find values:
+     */
+    private static final long SEPARATOR_PATTERN = compilePattern((byte) ';');
+
+    private int findNextSWAR(ByteBuffer bb, long pattern, int start, int limit) {
+        int i;
+        for (i = start; i <= limit - 8; i += 8) {
+            long word = bb.getLong(i);
+            int index = firstAnyPattern(word, pattern);
+            if (index < Long.BYTES) {
+                return i + index;
+            }
+        }
+        // Handle remaining bytes
+        for (; i < limit; i++) {
+            if (bb.get(i) == (byte) pattern) {
+                return i;
+            }
+        }
+        return limit; // delimiter not found
+    }
+
+    private static long compilePattern(byte value) {
+        return ((long) value << 56) | ((long) value << 48) | ((long) value << 40) | ((long) value << 32) |
+                ((long) value << 24) | ((long) value << 16) | ((long) value << 8) | (long) value;
+    }
+
+    private static int firstAnyPattern(long word, long pattern) {
+        final long match = word ^ pattern;
+        long mask = match - 0x0101010101010101L;
+        mask &= ~match;
+        mask &= 0x8080808080808080L;
+        return Long.numberOfTrailingZeros(mask) >>> 3;
```

Comment:
Here is shouldn't be the number of leading ones?

May that is why it hasn't @royvanrijn @gunnarmorling

Reply:
Weird, I thought by setting explicitly to LE would make the compatibility issues disappear. So running it on my machine would automatically mean it works on the target, although perhaps having a performance hit.

afk atm, I’ll check tomorrow, if somebody wants to fix it and tell me, be my guest 😂

Reply:
This method is one of the few cases I suggest to add a unit test!

Reply:
I have some local classes that test; problem is that I believe it works on my machine, just not on the target machine, I’ll check soon.

Reply:
Yeah, annoying, it runs fine locally (the code that was pushed), sigh. Kind of debugging in the dark haha... a challenge!

Reply:
Thinking about it twice, I'm wrong at https://github.com/gunnarmorling/1brc/pull/5/#discussion_r1440922218

let's say we have a byte[] data = { 0x01, 0x03 }
And we assume to have a short-based version of SWAR

reading the content of data with (a short) little-endian means: 

0x0301

which have the less significant part at the lower address,
hence the binary hex SWAR result obtained (I'm using the Netty algorithm, but here should be the same) will be

0x8000

and, in order to find out 0x03, we have to use the trailing zeros (here 8 + 7 = 15 -> 15/8 = 1) .

Which means that is fine as it is!